code/modules/surgery/bodyparts/_bodyparts.dm

```diff
@@ -198,6 +236,100 @@
 	update_disabled()
 	return update_bodypart_damage_state() || .
 
+
+/obj/item/bodypart/proc/check_wounding(woundtype, damage, wound_bonus, bare_wound_bonus)
+	// damage clothes if applicable, part of wounds
+
+	//TODO: do this better
+	// actually roll wounds if applicable
+	if(HAS_TRAIT(owner, TRAIT_EASYLIMBDISABLE))
+		damage *= 1.5
+
+	var/injury_roll = rand(1, round(damage ** WOUND_DAMAGE_EXPONENT))
+	//testing("Init roll| (1, [round(damage ** WOUND_DAMAGE_EXPONENT)]): [injury_roll]")
+	injury_roll += check_woundings_mods(woundtype, damage, wound_bonus, bare_wound_bonus)
+	//testing("Final roll: [injury_roll]")
+	var/list/wounds_checking
+
+	switch(woundtype)
+		if(WOUND_SHARP)
+			wounds_checking = WOUND_TYPE_CUT
+		if(WOUND_BRUTE)
+			wounds_checking = WOUND_TYPE_BONE
+		if(WOUND_BURN)
+			wounds_checking = WOUND_TYPE_BURN
+
+	//var/datum/wound/W
+	for(var/PW in wounds_checking)
+		var/datum/wound/possible_wound = PW
+		var/datum/wound/replaced_wound
+		for(var/datum/wound/existing_wound in wounds)
+			if(existing_wound.type in wounds_checking)
+				if(existing_wound.severity >= initial(possible_wound.severity))
+					return
+				else
+					replaced_wound = existing_wound
+
+		if(initial(possible_wound.threshold_minimum) < injury_roll)
+			if(replaced_wound)
+				replaced_wound.replace_wound(possible_wound)
+			else
+				var/datum/wound/new_wound = new possible_wound
+				new_wound.apply_wound(src)
+			return
+
+// try forcing a specific wound, but only if there isn't already a wound of that severity or greater for that type on this bodypart
+/obj/item/bodypart/proc/force_wound_upwards(specific_woundtype, smited = FALSE)
+	var/datum/wound/potential_wound = specific_woundtype
+	for(var/datum/wound/existing_wound in wounds)
```

Comment:
There should only be wounds in wounds list surely, there's few places where this should be untyped list?

Reply:
ack, missed those ones, cheers